### ISSUE TITLE
[Promtool] when checking for referenced files, `promtool check config` prints warnings for some checked files, fails check on others

### ISSUE THREAD

**Issue by zzehring** [prometheus/prometheus#9920]
## Proposal
**Use case. Why is this important?**

While working on #9909, I noticed that `promtool check config` has different results/behaviors when filepaths referenced in the config don't exist. The behavior differs depending on the section. 

For example, when evaluating rules files, a missing file will return an error that will fail the check:

https://github.com/prometheus/prometheus/blob/ec002ae0dc0352235c86d43e668564fdf0e05127/cmd/promtool/main.go#L348-L350

vs checking `config. ServiceDiscoveryConfigs.ServiceDiscoveryConfigs` of type `file.SDConfig`, which only prints a warning:

https://github.com/prometheus/prometheus/blob/ec002ae0dc0352235c86d43e668564fdf0e05127/cmd/promtool/main.go#L374-L391

In short, a missing rule file fails the entire check whereas a missing SD file just prints a warning. 

Unless there's more concern with some files over others, ideally these 2 scenarios should return similar results (i.e. missing files either print a warning or fail the check). I'd propose missing files would just be a warn. Then, we wouldn't need a `--syntax-only` flag for `check config`. From a CI perspective, I think it would make sense for the default behavior, as CI workspaces likely wouldn't have the referenced files in place. We could also add some sort of quiet flag to suppress the warning messages as well.

**Comment by JessicaGreben** [comment]
The places in the `promtool check config` that are warning instead of erroring are the sd config file checks.

After looking into this more, the current behavior is the desired behavior.  Currently, if rule files are provided to Prometheus, they are only registered during Prometheus startup/reloads. In contrast to service discovery files, which are read at runtime. We expect the rule files to be present on startup/reloads, but maybe sd files are written later and then read in dynamically at runtime.
